hack/internal/tools/ginkgo-top/ginkgo-top.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"cmp"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"slices"
+	"time"
+
+	gtypes "github.com/onsi/ginkgo/v2/types"
+	"gopkg.in/yaml.v3"
+)
+
+type SpecDescription struct {
+	Suite              string
+	ContainerHierarchy []string
+	Name               string
+	Duration           time.Duration
+	Events             []Event
+}
+
+type Event struct {
+	Message  string
+	Location string
+	Duration time.Duration
+}
+
+var (
+	reportPath = flag.String("i", "ginkgo.json", "input file name")
+	count      = flag.Uint("count", 0, "number of items")
+	noEvents   = flag.Bool("noEvents", false, "don't show events")
```

Comment:
```suggestion
	withEvents   = flag.Bool("withEvents", false, "show events")
```
to avoid double negation in code, but here I would also suggest to drop the param completely, and assume `withEvents=true`.

Reply:
changed